packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,12 +352,261 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+    }
+
+    private async fetchSnapshotWithMultiPartRequestCore(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions) {
         const storageToken = await this.getStorageToken(tokenFetchOptions, "TreesLatest");
 
         // TODO: This snapshot will return deltas, which we currently aren't using. We need to enable this flag to go down the "optimized"
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
-        const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
+        const url = `${this.snapshotUrl}/trees/latest?ump=1`;
+        const formBoundary = uuid();
+        const crlf = "\r\n";
+        let postBody = `--${formBoundary}${crlf}Authorization: Bearer ${storageToken}${crlf}X-HTTP-Method-Override: GET${crlf}`;
+        Object.entries(snapshotOptions).forEach(([key, value]) => {
+            postBody += `${key}: ${value}${crlf}`;
+        });
+        postBody += `${crlf}--${formBoundary}--`;
+        const headers = {
+            "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+        };
+        // This event measures only successful cases of getLatest call (no tokens, no retries).
+        const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
+            const response = await fetchHelper<IOdspSnapshot>(
+                url,
+                {
+                    body: postBody,
+                    headers,
+                    method: "POST",
+                });
+            const content = response.content;
+            event.end({
+                trees: content.trees?.length ?? 0,
+                blobs: content.blobs?.length ?? 0,
+                ops: content.ops?.length ?? 0,
+                headers: Object.keys(headers).length !== 0 ? true : undefined,
+                sprequestguid: response.headers.get("sprequestguid"),
+                sprequestduration: TelemetryLogger.numberFromString(response.headers.get("sprequestduration")),
+                contentsize: TelemetryLogger.numberFromString(response.headers.get("content-length")),
+                bodysize: TelemetryLogger.numberFromString(response.headers.get("body-size")),
+            });
+            return {
+                snapshot: content,
+                // There are some scenarios in ODSP where we cannot cache, trees/latest will explicitly tell us when we cannot cache using an HTTP response header.
+                canCache: response.headers.get("disablebrowsercachingofusercontent") !== "true",
+            };
+        });
+
+        assert(this._snapshotCacheEntry === undefined);
+        this._snapshotCacheEntry = {
+            file: this.fileEntry,
+            type: "snapshot",
+            key: "",
+        };
+
+        // There maybe no snapshot - TreesLatest would return just ops.
```

Comment:
This is probably shareable code (with the old code paths).. If the idea is to remove the old code after veritying this, then this is probably ok.

Reply:
Yes, we will remove the other code once we get confidence in POST call.